woocommerce/woocommerce
# Add requires_jpc field to determine plugins that require JPC

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #52887 .

This PR adds `requires_jpc` field to Jetpack and WooCommerce Payments. This field indicates that the plugin requires a Jetpack connection. When the plugin is selected during the onboarding, users will be redirected to the Jetpack connection page after the plugin is installed.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a new JN site. 
2. Before starting the core profiler, navigate to `WooCommerce -> Settings -> Advanced -> WooCommerce.com` and turn off `Display suggestions within WooCommerce`
3. Start the core profiler. If you're already on the first page, make sure to refresh it.
4. Proceed to the extensions page and select WooCommerce Payments.
5. Continue
6. You should be redirected to the Jetpack connection page.
7. Repeat the process with Jetpack and Jetpack + WooCommerce Payments.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

Add requires_jpc field to determine plugins that require JPC

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
